File: HealthTracker/Models/WeightRecord.swift
//
//  WeightRecord.swift
//  HealthTracker
//
//  体重记录数据模型
//

import Foundation

struct WeightRecord: Identifiable, Codable, Equatable {
    let id: UUID
    var weight: Double      // 体重 (kg)
    var date: Date
    var note: String?
    
    init(id: UUID = UUID(), weight: Double, date: Date = Date(), note: String? = nil) {
        self.id = id
        self.weight = weight
        self.date = date
        self.note = note
    }
    
    var formattedWeight: String {
        String(format: "%.1f kg", weight)
    }
    
    var formattedDate: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        formatter.locale = Locale(identifier: "zh_CN")
        return formatter.string(from: date)
    }
}

File: android/app/src/main/java/com/wanxx/healthtracker/data/WeightRecord.kt
package com.wanxx.healthtracker.data

import java.util.Date
import java.util.UUID

data class WeightRecord(
    val id: String = UUID.randomUUID().toString(),
    val weight: Double,
    val date: Date = Date(),
    val note: String? = null
) {
    fun formattedWeight(): String = "%.1f kg".format(weight)

    fun formattedDate(): String {
        val sdf = java.text.SimpleDateFormat("yyyy/MM/dd HH:mm", java.util.Locale.CHINA)
        return sdf.format(date)
    }
}
